### ISSUE TITLE
typos (group<->user) in twisted.words.service

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#11161]
|<img alt="j2a's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| j2a reported|
|-|-|
|Trac ID|trac#1655|
|Type|defect|
|Created|2006-04-20 13:17:31Z|

Couple of typos here... patch included:

```
--- service.py.orig     2005-10-22 11:08:27.000000000 +0700
+++ service.py  2006-04-20 20:07:31.301166500 +0700
@@ -1049,7 +1049,7 @@
     createGroupOnRequest = False
     createUserOnRequest = True
 
-    def lookupGroup(self, name):
+    def lookupUser(self, name):
         raise NotImplementedError
 
 
@@ -1158,7 +1158,7 @@
 
 
     def addGroup(self, group):
-        if group.name in self.users:
+        if group.name in self.groups:
             return defer.fail(failure.Failure(ewords.DuplicateGroup()))
         self.groups[group.name] = group
         return defer.succeed(group)
```

<details><summary>Searchable metadata</summary>

```
trac-id__1655 1655
type__defect defect
reporter__j2a j2a
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__words words
keywords__ 
time__1145539051000000 1145539051000000
changetime__1147635787000000 1147635787000000
version__ 
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @teratorn|
|-|-|

Applied and unit test added in user-group-collision-1655.  Please review.

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)|@radix set owner to @radix<br>@radix set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)|@radix set owner to @exarkun<br>@radix set status to `new`|
|-|-|

+1.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set status to `closed`|
|-|-|

(In [16808]) Merge user-group-collision-1655

Author: j2a, exarkun
Reviewer: radix
Fixes [#11161](https://github.com/twisted/twisted/issues/11161)

Add a test for adding a user and group with the same name and
fix InMemoryWordsRealm so that this works.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
